cudax/CMakeLists.txt

```
@@ -13,6 +13,9 @@ endif()
 
 project(cudax LANGUAGES CUDA CXX)
 
+find_package(Thrust REQUIRED CONFIG)
```

This should use the `NO_DEFAULT_PATH` `HINTS` option to ensure that we'll pulling Thrust from the repo.
```suggestion
find_package(Thrust REQUIRED CONFIG
  NO_DEFAULT_PATH # Only check the explicit path in HINTS:
  HINTS "${CCCL_SOURCE_DIR}/thrust"
)
```

---

This block should also be moved into `_cn_add_target_to_target_list` in `cudaxBuildCompilerTargets` [here](https://github.com/alliepiper/cccl/blob/sccache_investigation/cudax/cmake/cudaxBuildTargetList.cmake#L128). The Thrust package and target should be found in that function, and the `cudax::Thrust` target added to the `target_link_libraries(${target_name} INTERFACE ...)` list. This ensures that the cudax targets we use across the build will all include the `Thrust` dependency.